src/main/java/org/biouno/unochoice/model/ScriptlerScript.java

```diff
@@ -131,7 +131,8 @@ public GroovyScript toGroovyScript() {
         if (scriptler == null) {
             throw new RuntimeException("Missing required scriptler!");
         }
-        return new GroovyScript(new SecureGroovyScript(scriptler.script, true, null), null);
+        boolean isSandbox = ScriptHelper.isApproved(scriptler.script) ? false : true;
+        return new GroovyScript(new SecureGroovyScript(scriptler.script, isSandbox, null), null);
```

Comment:
A small fix with great benefit to users! Thanks @rafeeqpv , I will just need to i) check the issue to remind myself of the context, and ii) see if I can write a quick test so we don't have a regression in the future. Maybe iii) check if there's no security issue enabling it in Jenkins plug-ins. Will try to include this issue in the next release.

Reply:
On iii), the code with `isSandbox=true` was added while fixing SECURITY-2008 issue in this commit: https://github.com/jenkinsci/active-choices-plugin/commit/62fb22b2d21d3bd2dfeaa051f283686c65dd6f7c

So definitely need care to merge it, so we don't re-introduce the security bug (which would end up removing the plug-in from update centre to all users)

Reply:
Reading https://www.jenkins.io/doc/book/managing/script-approval/.

I think SECURITY-2008 because we were always evaluating the Groovy scripts in a non-sandbox environment and without approval.

At the moment we are always evaluating in a sandbox environment.

The change here would change it so that if a script has not been approved, then it would be sandboxed. But if already approved by an administrator, then the script would be evaluated without the Groovy sandbox.

I think it follows the model proposed in the Groovy Sandbox + Script Security, and doesn't add any security issues. So all good for iii) IMO.